python/ray/dag/compiled_dag_node.py

```diff
@@ -442,12 +433,18 @@ def _preprocess(self) -> None:
                     )
                 self.actor_task_count[actor_handle._actor_id] += 1
 
-                if (
-                    isinstance(dag_node.type_hint, TorchTensorType)
-                    and dag_node.type_hint.transport == "nccl"
-                ):
+                if dag_node.type_hint.requires_nccl():
                     # Add all writers to the NCCL group.
                     nccl_actors.add(actor_handle)
+            elif isinstance(dag_node, InputNode):
+                if dag_node.type_hint.requires_nccl():
+                    raise ValueError("DAG inputs cannot be transferred via NCCL")
```

Comment:
out of curiosity: why can't DAG inputs be transferred via NCCL? Is it because (1) The DAG inputs must come from the driver process, and the driver process cannot run on a GPU and (2) NCCL is used to transfer data between GPUs?

Reply:
Yes, it's because of that. Let me update the message to make that clearer.